### PR TITLE
feat: default repack encryption

### PR DESCRIPTION
**Pull request by JGuinegagne**
*Issue #, if available:*
AppSec finding affecting SageMaker JumpStart TT: V497918765

*Description of changes:*
This PR changes the default behavior of the SDK when uploading artifacts to the S3 bucket of the `Session`. Previously, if no `kms_key` argument was specified, the SDK uploaded artifact to `S3` without server-side encryption. Now, the artifact on `S3` is uploaded with server-side encryption enabled, and `S3` uses the AWS managed KMS key for S3 in the customer account.

According to KMS team and AppSec, there is no downside to enabling server-side encryption in `S3` for customers: any `S3:getObject` call to such objects will be automatically decrypted by `S3` if the requester has access to the object.  If a customer wishes to disable the server-side encryption for their artifacts anyway (for example to use the public object url), we provide a `SessionSettings` class as an optional constructor argument that lets them override the default behavior.

This `SessionSettings` class is meant to be extended to allow customers to parametrize their SageMaker `Session`.

*Testing done:*
- manually updated the `sagemaker` package in JumpStart [server](https://github.com/aws/sagemaker-jumpstart-python-server) dependency with matching change, and verified that deploying an endpoint lead to the model tarball in `S3` to use server-side encryption with the default AWS managed KMS key for S3. 

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/unit/test_fw_utils.py [hunk 1]
```diff
@@ -93,6 +94,40 @@ def test_tar_and_upload_dir_s3_with_kms(utils, sagemaker_session):
     obj.upload_file.assert_called_with(utils.create_tar_file(), ExtraArgs=extra_args)
 
 
+@patch("sagemaker.utils")
+def test_tar_and_upload_dir_s3_without_kms_no_settings(utils, sagemaker_session):
```
Comment (mufaddal-rohawala) [inline tests/unit/test_fw_utils.py:98]: nit: should this be `test_tar_and_upload_dir_s3_with_kms_no_settings` ?

File: .gitignore [hunk 2]
```diff
@@ -27,4 +27,5 @@ venv/
 *.swp
 .docker/
 env/
-.vscode/
\ No newline at end of file
+.vscode/
+.python-version
```
[the remark this answers is not in the document]
Comment (JGuinegagne) [reply .gitignore:31]: I use pyenv which creates this file in the project. I would prefer to keep the gitignore entry to ensure I don't commit it in the future.

File: tests/unit/test_fw_utils.py (hunk 1 above)
Comment (JGuinegagne) [reply tests/unit/test_fw_utils.py:98]: here I'm testing that even when we don't pass a `kms_key` argument the artifact uploaded to `S3` is encrypted at rest with the default AWS managed KMS key for `S3`.

How about renaming `test_tar_and_upload_dir_s3_no_custom_kms_no_settings`?

**Review by mufaddal-rohawala — APPROVED** [review]

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (shreyapandit) [reply .gitignore:31]: Makes sense.

**Review by shreyapandit — APPROVED** [review]

**Review by evakravi — APPROVED** [review]

**Pull request merged**